tests/lit-tests/signed-integer-overflow.ispc

```diff
@@ -0,0 +1,54 @@
+// This test checks for the use of the no signed wrap (`nsw`) attribute for arithmetic
+// expressions which may suffer from overflow.
+// Needed for induction variable simplification dicussed in
+// [GitHub issue ISPC/#2460](https://github.com/ispc/ispc/issues/2460).
+
+// RUN: %{ispc} %s -o %t.ll -O2 --pic --emit-llvm-text --target=avx2 --arch=x86-64
```

Comment:
1. You need to check the IR after the front-end (dump with `--debug-phase=first`), not the IR after full optimization at -O2.
2. What `--pic` switch is doing here?
3. What target is `avx2`?? First of all we deprecated usch format of target naming and use full format, second it should be `host` and should not run only on x86, it should run on any target and should not depend on the target.
4. Use `--nostdlib` switch to run tests faster.
5. Use pipe to feed the output to `FileCheck` instead of creating temporary files.

Reply:
> 1. You need to check the IR after the front-end (dump with `--debug-phase=first`), not the IR after full optimization at -O2.
> 2. What `--pic` switch is doing here?
> 3. What target is `avx2`?? First of all we deprecated usch format of target naming and use full format, second it should be `host` and should not run only on x86, it should run on any target and should not depend on the target.
> 4. Use `--nostdlib` switch to run tests faster.
> 5. Use pipe to feed the output to `FileCheck` instead of creating temporary files.
1.  Make sense!
2.  It's not needed for the lit test(s); will remove. For CPU kernels compiled via the ISPCRT API, I believe this flag is needed, so I've just made a habit of using it.
3.  The [example](https://ispc.godbolt.org/z/Gfre6sG3q) for the [issue](https://github.com/ispc/ispc/issues/2460) this patch addresses is built using this flag, so I've been addressing the issue matching this flag. Do you suggest specifying a specific width always? We should advertise that this method shouldn't be used.
4.  Will do.
5.  Sometimes this way of doing thing requires `--nowrap` which I find messier. Also, having the compilation and the lit test in separate commands makes it clearer what's broken when a test fails.

Reply:
> 3. The [example](https://ispc.godbolt.org/z/Gfre6sG3q) for the [issue](https://github.com/ispc/ispc/issues/2460) this patch addresses is built using this flag, so I've been addressing the issue matching this flag. Do you suggest specifying a specific width always? We should advertise that this method shouldn't be used.

Short version of targets are declared deprecated long time ago.

> 5. Sometimes this way of doing thing requires `--nowrap` which I find messier. Also, having the compilation and the lit test in separate commands makes it clearer what's broken when a test fails.

`--nowrap` affects the output for error messages and warnings. It doesn't affect dumps and what is streamed to the pipe (i.e. `-o -`).